### PR TITLE
tmp fix nullptr reference - this will be cleaned up soon in a better …

### PR DESCRIPTION
**Pull request by hkernbach**
…way, as soon as we can get rid of the old non refactored code

### Scope & Purpose

*(Please describe the changes in this PR for reviewers, motivation, rationale - **mandatory**)*

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Aql/TraversalExecutor.h [hunk 1]
```diff
@@ -180,7 +180,7 @@ class TraversalExecutorInfos {
       TraversalExecutorInfosHelper::OutputName type) const;
 
  private:
-  std::unique_ptr<traverser::Traverser>
+  std::shared_ptr<traverser::Traverser>
```
Comment (mchacki) [inline arangod/Aql/TraversalExecutor.h:183]: Why is this made a shared_ptr now?
We only hand out a raw pointer, which we can from a unique_ptr as well.
I think this specific change should be rolled back.

File: arangod/Aql/TraversalExecutor.cpp [hunk 2]
```diff
@@ -194,7 +194,10 @@ TraversalExecutorInfos::traversalEnumerator() const {
 }
```
Comment (mchacki) [inline arangod/Aql/TraversalExecutor.cpp:194]: The code above this does have exactly the same issue!
We call both of the Methods next to each other, but only one actually has a pointer (depending on the "isRefactor" state)
So I think to make this correct, all changes in the PR need to be applied for the REFACTOR _traversalEnumerator as well.

File: arangod/Aql/TraversalExecutor.h (hunk 1 above)
Comment (hkernbach) [reply arangod/Aql/TraversalExecutor.h]: put back to unique ptr

File: arangod/Aql/TraversalExecutor.cpp [hunk 3]
```diff
@@ -1178,6 +1184,15 @@ bool TraversalExecutor::initTraverser(AqlItemBlockInputRange& input) {
...
+traverser::Traverser* TraversalExecutor::traverser() {
+  TRI_ASSERT(_traverser != nullptr);
+  return _traverser;
+}
+
+arangodb::graph::TraversalEnumerator* TraversalExecutor::traversalEnumerator() {
+  TRI_ASSERT(_traversalEnumerator != nullptr);
+  return _traversalEnumerator;
+}
```
Comment (mchacki) [inline arangod/Aql/TraversalExecutor.cpp:1198]: I would have used references here, but this is good as well :+1:

**Review by mchacki — APPROVED** [review]
LGTM issue fixed!

**Pull request merged**